### PR TITLE
add building and uploading windows installer to CI

### PR DESCRIPTION
**Pull request by llogiq**
This adds a plain WiX installer configuration with a suitable license, name & homepage, and is my attempt to extend the release workflow to build and ship it.

Open tasks:

* [x] Test the release process @brokad
* [ ] ~Add a link to the website~
* [ ] ~code-sign the installer (?) This would require a key signed by microsoft and remove the intimidating warning about installing an unsigned binary from the install process~

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -169,3 +169,26 @@ jobs:
...
+      - id: copy-target
+        run: |
+          mkdir synth/target
+          mkdir synth/target/release
+          copy target/release/synth.exe synth/target/release
+      - id: wix
+        run: |
+          cd synth
+          cargo wix
+      - name: Upload Windowd installer
```
Comment (brokad) [inline .github/workflows/release.yml:185]:
```suggestion
      - name: Upload Windows installer
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -169,3 +169,26 @@ jobs:
...
+        id: upload-windows-installer
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.release.outputs.upload_url }}
+          asset_path: ./synth/target/wix/synth-${{ steps.tag.outputs.tag }}-x86_64.msi
```
Comment (brokad) [inline .github/workflows/release.yml:192]: We should also document this route in the installation pages of the docs.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -169,3 +169,26 @@ jobs:
...
+          cargo +nightly install cargo-wix
+      - id: copy-target
+        run: |
+          mkdir synth/target
+          mkdir synth/target/release
+          copy target/release/synth.exe synth/target/release
+      - id: wix
+        run: |
+          cd synth
+          cargo wix
```
Comment (brokad) [inline .github/workflows/release.yml:184]: The `main.wxs` is empty, which I suppose means everything is left to default?

When I did this locally, that file was populated with boilerplate and in particular the installer offered to add the binary to the path. Will that be done with defaults as well?

Comment (llogiq) [reply .github/workflows/release.yml:184]: Yes -- At least that was what worked for me.

**Comment by christoshadjiaslanis** [comment]
LGTM. Gotta update the links now to point to the `msi`.

**Pull request merged**